### ISSUE TITLE
make test for 6.4.5 fails at t/mojolicious/commands.t

### ISSUE THREAD

**Issue by mghicks** [kraih/mojo#908]
cpanm Mojolicious fails for 6.4.5, so I did a little digging--excerpts below.

```
$ make test
PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t t/*/*.t
t/mojo/asset.t ............................. ok     
...
t/mojolicious/commands.t ................... 1/? Problem uploading file "t/mojolicious/commands.t": cannotconnect
# Tests were run but no plan was declared and done_testing() was not seen.
# Looks like your test exited with 255 just after 34.
t/mojolicious/commands.t ................... Dubious, test returned 255 (wstat 65280, 0xff00)
All 34 subtests passed 

Test Summary Report
-------------------
t/mojolicious/commands.t                 (Wstat: 65280 Tests: 34 Failed: 0)
  Non-zero exit status: 255
  Parse errors: No plan found in TAP output
Files=86, Tests=10391, 29 wallclock secs ( 1.09 usr  0.26 sys + 16.88 cusr  2.39 csys = 20.62 CPU)
Result: FAIL
Failed 1/86 test programs. 0/10391 subtests failed.
make: *** [test_dynamic] Error 255

```

But it passes cleanly with prove -Iblib and fails the same way with prove -Ilib:

```
$ prove -Iblib -mv t/mojolicious/commands.t
t/mojolicious/commands.t .. 
ok 1 - right mode
ok 2 - unchanged
ok 3 - right environment
ok 4 - right environment
ok 5 - right environment
ok 6 - no environment
ok 7 - right reference
ok 8 - not a CODE reference
ok 9 - right reference
ok 10 - right class
ok 11 - right reference
ok 12 - right result
ok 13 - right result
ok 14 - right reference
ok 15 - no mode
ok 16 - right namespaces
ok 17 - has a description
ok 18 - has a message
ok 19 - has a hint
ok 20 - right output
ok 21 - description has been picked up
ok 22 - fake command has been ignored
ok 23 - right output
ok 24 - right output
ok 25 - right output
ok 26 - has a description
ok 27 - has usage information
ok 28 - right output
ok 29 - right output
ok 30 - right output
ok 31 - has a description
ok 32 - has usage information
ok 33 - has a description
ok 34 - has usage information
ok 35 - right output
ok 36 - has a description
ok 37 - has usage information
ok 38 - has a description
ok 39 - has usage information
ok 40 - right output
ok 41 - right namespaces
ok 42 - has a description
ok 43 - has a message
ok 44 - has a hint
ok 45 - right output
ok 46 - has a description
ok 47 - has usage information
ok 48 - right output
ok 49 - script exists
ok 50 - application class exists
ok 51 - controller exists
ok 52 - test exists
ok 53 - static file exists
ok 54 - layout exists
ok 55 - template exists
ok 56 - has a description
ok 57 - has usage information
ok 58 - right output
ok 59 - app exists
ok 60 - has a description
ok 61 - has usage information
ok 62 - right output
ok 63 - Makefile.PL exists
ok 64 - has a description
ok 65 - has usage information
ok 66 - right output
ok 67 - class exists
ok 68 - test exists
ok 69 - Makefile.PL exists
ok 70 - has a description
ok 71 - has usage information
ok 72 - has a description
ok 73 - has usage information
ok 74 - has a description
ok 75 - has usage information
ok 76 - has a description
ok 77 - has usage information
ok 78 - right output
ok 79 - not verbose
ok 80 - right output
ok 81 - verbose
ok 82 - has a description
ok 83 - has usage information
ok 84 - has a description
ok 85 - has usage information
ok 86 - right output
ok 87 - right output
1..87
ok
All tests successful.
Files=1, Tests=87,  1 wallclock secs ( 0.03 usr  0.00 sys +  0.30 cusr  0.03 csys =  0.36 CPU)
Result: PASS

$ prove -Ilib -mv t/mojolicious/commands.t
t/mojolicious/commands.t .. 
ok 1 - right mode
ok 2 - unchanged
ok 3 - right environment
ok 4 - right environment
ok 5 - right environment
ok 6 - no environment
ok 7 - right reference
ok 8 - not a CODE reference
ok 9 - right reference
ok 10 - right class
ok 11 - right reference
ok 12 - right result
ok 13 - right result
ok 14 - right reference
ok 15 - no mode
ok 16 - right namespaces
ok 17 - has a description
ok 18 - has a message
ok 19 - has a hint
ok 20 - right output
ok 21 - description has been picked up
ok 22 - fake command has been ignored
ok 23 - right output
ok 24 - right output
ok 25 - right output
ok 26 - has a description
ok 27 - has usage information
ok 28 - right output
ok 29 - right output
ok 30 - right output
ok 31 - has a description
ok 32 - has usage information
ok 33 - has a description
ok 34 - has usage information
Problem uploading file "t/mojolicious/commands.t": cannotconnect
# Tests were run but no plan was declared and done_testing() was not seen.
# Looks like your test exited with 255 just after 34.
Dubious, test returned 255 (wstat 65280, 0xff00)
All 34 subtests passed 

Test Summary Report
-------------------
t/mojolicious/commands.t (Wstat: 65280 Tests: 34 Failed: 0)
  Non-zero exit status: 255
  Parse errors: No plan found in TAP output
Files=1, Tests=34,  1 wallclock secs ( 0.02 usr  0.00 sys +  0.26 cusr  0.03 csys =  0.31 CPU)
Result: FAIL
```

**Comment by kraih** [comment]
Already fixed. https://github.com/kraih/mojo/commit/c33f844c68e2dfe492ff1c1f8e7f869a66f4587e

**Comment by mghicks** [comment]
```
$ perl -V
Summary of my perl5 (revision 5 version 18 subversion 2) configuration:
   
  Platform:
    osname=linux, osvers=3.2.0-58-generic, archname=x86_64-linux-gnu-thread-multi
    uname='linux brownie 3.2.0-58-generic #88-ubuntu smp tue dec 3 17:37:58 utc 2013 x86_64 x86_64 x86_64 gnulinux '
    config_args='-Dusethreads -Duselargefiles -Dccflags=-DDEBIAN -D_FORTIFY_SOURCE=2 -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -Dldflags= -Wl,-Bsymbolic-functions -Wl,-z,relro -Dlddlflags=-shared -Wl,-Bsymbolic-functions -Wl,-z,relro -Dcccdlflags=-fPIC -Darchname=x86_64-linux-gnu -Dprefix=/usr -Dprivlib=/usr/share/perl/5.18 -Darchlib=/usr/lib/perl/5.18 -Dvendorprefix=/usr -Dvendorlib=/usr/share/perl5 -Dvendorarch=/usr/lib/perl5 -Dsiteprefix=/usr/local -Dsitelib=/usr/local/share/perl/5.18.2 -Dsitearch=/usr/local/lib/perl/5.18.2 -Dman1dir=/usr/share/man/man1 -Dman3dir=/usr/share/man/man3 -Dsiteman1dir=/usr/local/man/man1 -Dsiteman3dir=/usr/local/man/man3 -Duse64bitint -Dman1ext=1 -Dman3ext=3perl -Dpager=/usr/bin/sensible-pager -Uafs -Ud_csh -Ud_ualarm -Uusesfio -Uusenm -Ui_libutil -Uversiononly -DDEBUGGING=-g -Doptimize=-O2 -Duseshrplib -Dlibperl=libperl.so.5.18.2 -des'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=define, usemultiplicity=define
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=define, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fstack-protector -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2 -g',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fstack-protector -fno-strict-aliasing -pipe -I/usr/local/include'
    ccversion='', gccversion='4.8.2', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -fstack-protector -L/usr/local/lib'
    libpth=/usr/local/lib /lib/x86_64-linux-gnu /lib/../lib /usr/lib/x86_64-linux-gnu /usr/lib/../lib /lib /usr/lib
    libs=-lgdbm -lgdbm_compat -ldb -ldl -lm -lpthread -lc -lcrypt
    perllibs=-ldl -lm -lpthread -lc -lcrypt
    libc=, so=so, useshrplib=true, libperl=libperl.so.5.18.2
    gnulibc_version='2.19'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
    cccdlflags='-fPIC', lddlflags='-shared -L/usr/local/lib -fstack-protector'


Characteristics of this binary (from libperl): 
  Compile-time options: HAS_TIMES MULTIPLICITY PERLIO_LAYERS
                        PERL_DONT_CREATE_GVSV
                        PERL_HASH_FUNC_ONE_AT_A_TIME_HARD
                        PERL_IMPLICIT_CONTEXT PERL_MALLOC_WRAP
                        PERL_PRESERVE_IVUV PERL_SAWAMPERSAND USE_64_BIT_ALL
                        USE_64_BIT_INT USE_ITHREADS USE_LARGE_FILES
                        USE_LOCALE USE_LOCALE_COLLATE USE_LOCALE_CTYPE
                        USE_LOCALE_NUMERIC USE_PERLIO USE_PERL_ATOF
                        USE_REENTRANT_API
  Locally applied patches:
	DEBPKG:debian/cpan_definstalldirs - Provide a sensible INSTALLDIRS default for modules installed from CPAN.
	DEBPKG:debian/db_file_ver - http://bugs.debian.org/340047 Remove overly restrictive DB_File version check.
	DEBPKG:debian/doc_info - Replace generic man(1) instructions with Debian-specific information.
	DEBPKG:debian/enc2xs_inc - http://bugs.debian.org/290336 Tweak enc2xs to follow symlinks and ignore missing @INC directories.
	DEBPKG:debian/errno_ver - http://bugs.debian.org/343351 Remove Errno version check due to upgrade problems with long-running processes.
	DEBPKG:debian/libperl_embed_doc - http://bugs.debian.org/186778 Note that libperl-dev package is required for embedded linking
	DEBPKG:fixes/respect_umask - Respect umask during installation
	DEBPKG:debian/writable_site_dirs - Set umask approproately for site install directories
	DEBPKG:debian/extutils_set_libperl_path - EU:MM: Set location of libperl.a to /usr/lib
	DEBPKG:debian/no_packlist_perllocal - Don't install .packlist or perllocal.pod for perl or vendor
	DEBPKG:debian/prefix_changes - Fiddle with *PREFIX and variables written to the makefile
	DEBPKG:debian/fakeroot - Postpone LD_LIBRARY_PATH evaluation to the binary targets.
	DEBPKG:debian/instmodsh_doc - Debian policy doesn't install .packlist files for core or vendor.
	DEBPKG:debian/ld_run_path - Remove standard libs from LD_RUN_PATH as per Debian policy.
	DEBPKG:debian/libnet_config_path - Set location of libnet.cfg to /etc/perl/Net as /usr may not be writable.
	DEBPKG:debian/mod_paths - Tweak @INC ordering for Debian
	DEBPKG:debian/module_build_man_extensions - http://bugs.debian.org/479460 Adjust Module::Build manual page extensions for the Debian Perl policy
	DEBPKG:debian/prune_libs - http://bugs.debian.org/128355 Prune the list of libraries wanted to what we actually need.
	DEBPKG:fixes/net_smtp_docs - [rt.cpan.org #36038] http://bugs.debian.org/100195 Document the Net::SMTP 'Port' option
	DEBPKG:debian/perlivp - http://bugs.debian.org/510895 Make perlivp skip include directories in /usr/local
	DEBPKG:debian/cpanplus_definstalldirs - http://bugs.debian.org/533707 Configure CPANPLUS to use the site directories by default.
	DEBPKG:debian/cpanplus_config_path - Save local versions of CPANPLUS::Config::System into /etc/perl.
	DEBPKG:debian/deprecate-with-apt - http://bugs.debian.org/702096 Point users to Debian packages of deprecated core modules
	DEBPKG:debian/squelch-locale-warnings - http://bugs.debian.org/508764 Squelch locale warnings in Debian package maintainer scripts
	DEBPKG:debian/skip-upstream-git-tests - Skip tests specific to the upstream Git repository
	DEBPKG:debian/patchlevel - http://bugs.debian.org/567489 List packaged patches for 5.18.2-2ubuntu1 in patchlevel.h
	DEBPKG:debian/skip-kfreebsd-crash - http://bugs.debian.org/628493 [perl #96272] Skip a crashing test case in t/op/threads.t on GNU/kFreeBSD
	DEBPKG:fixes/document_makemaker_ccflags - http://bugs.debian.org/628522 [rt.cpan.org #68613] Document that CCFLAGS should include $Config{ccflags}
	DEBPKG:debian/find_html2text - http://bugs.debian.org/640479 Configure CPAN::Distribution with correct name of html2text
	DEBPKG:debian/hurd_test_skip_stack - http://bugs.debian.org/650175 Disable failing GNU/Hurd tests dist/threads/t/stack.t
	DEBPKG:fixes/manpage_name_Test-Harness - http://bugs.debian.org/650451 [rt.cpan.org #73399] cpan/Test-Harness: add NAME headings in modules with POD
	DEBPKG:debian/makemaker-pasthru - http://bugs.debian.org/660195 [rt.cpan.org #28632] Make EU::MM pass LD through to recursive Makefile.PL invocations
	DEBPKG:debian/perl5db-x-terminal-emulator.patch - http://bugs.debian.org/668490 Invoke x-terminal-emulator rather than xterm in perl5db.pl
	DEBPKG:debian/cpan-missing-site-dirs - http://bugs.debian.org/688842 Fix CPAN::FirstTime defaults with nonexisting site dirs if a parent is writable
	DEBPKG:fixes/memoize_storable_nstore - [rt.cpan.org #77790] http://bugs.debian.org/587650 Memoize::Storable: respect 'nstore' option not respected
	DEBPKG:fixes/net_ftp_failed_command - [rt.cpan.org #37700] http://bugs.debian.org/491062 Net::FTP: cope gracefully with a failed command
	DEBPKG:fixes/perlbug-patchlist - [3541c11] http://bugs.debian.org/710842 [perl #118433] Make perlbug look up the list of local patches at run time
	DEBPKG:fixes/module_metadata_security_doc - [68cdd4b] CVE-2013-1437 documentation fix
	DEBPKG:fixes/module_metadata_taint_fix - [bff978f] http://bugs.debian.org/722210 [rt.cpan.org #88576] untaint version, if needed, in Module::Metadata
	DEBPKG:fixes/IPC-SysV-spelling - http://bugs.debian.org/730558 [rt.cpan.org #86736] Fix spelling of IPC_CREAT in IPC-SysV documentation
	DEBPKG:fixes/fix-undef-source -
  Built under linux
  Compiled at Mar 27 2014 18:30:28
  %ENV:
    PERL5LIB="/home/hickmat/perl5/lib/perl5"
    PERL_LOCAL_LIB_ROOT="/home/hickmat/perl5"
    PERL_MB_OPT="--install_base "/home/hickmat/perl5""
    PERL_MM_OPT="INSTALL_BASE=/home/hickmat/perl5"
  @INC:
    /home/hickmat/perl5/lib/perl5/x86_64-linux-gnu-thread-multi
    /home/hickmat/perl5/lib/perl5
    /etc/perl
    /usr/local/lib/perl/5.18.2
    /usr/local/share/perl/5.18.2
    /usr/lib/perl5
    /usr/share/perl5
    /usr/lib/perl/5.18
    /usr/share/perl/5.18
    /usr/local/lib/site_perl
    .

```

**Comment by mghicks** [comment]
Thanks!
